src/shared_game_state.rs

```diff
@@ -67,6 +67,49 @@ impl TimingMode {
     }
 }
 
+#[derive(PartialEq, Eq, Copy, Clone)]
+pub enum GameDifficulty {
+    Easy,
+    Normal,
+    Hard,
+}
+
+impl GameDifficulty {
+    pub fn from_index(index: usize) -> GameDifficulty {
+        match index {
+            0 => GameDifficulty::Easy,
+            1 => GameDifficulty::Normal,
+            2 => GameDifficulty::Hard,
+            _ => unreachable!(),
+        }
+    }
+
+    pub fn from_save_value(val: u8) -> GameDifficulty {
+        match val {
+            0 => GameDifficulty::Normal,
+            2 => GameDifficulty::Easy,
+            4 => GameDifficulty::Hard,
+            _ => unreachable!(),
+        }
+    }
+
+    pub fn to_save_value(self) -> u8 {
+        match self {
+            GameDifficulty::Normal => 0,
+            GameDifficulty::Easy => 2,
+            GameDifficulty::Hard => 4,
+        }
+    }
+
+    pub fn get_skinsheet_offset(difficulty: GameDifficulty) -> u16 {
+        match difficulty {
+            GameDifficulty::Easy => 1,   // Yellow Quote
+            GameDifficulty::Normal => 0, // Good Quote
+            GameDifficulty::Hard => 2,   // Human Quote
+        }
+    }
+}
+
```

Comment:
I'm not 100% on this, but I think it looks cleaner: The enum can be defined for each, `Easy = 2,` etc so you can cast difficulty -> number and can add `FromPrimitive` so the reverse can be done. I don't think it should panic either if the save file has a weird number for difficulty, instead just setting it to normal and continuing. 

```
#[derive(PartialEq, Eq, Copy, Clone, FromPrimitive)]
pub enum GameDifficulty {
    Easy = 2,
    Normal = 0,
    Hard = 4,
}

pub fn from_save_value(val: u8) -> GameDifficulty {
        return num_traits::FromPrimitive::from_u8(val).unwrap_or(GameDifficulty::Normal);
}
```

Reply:
I think I'll refactor this enum so it follows CS+'s weird format. Will eliminate the need of a lot of methods here.

Reply:
That they save it as 0/2/4 which just happens to be the number of sprite sheet entries (if you count mimiga mask sprites) suddenly makes sense.